src/inference/src/ie_network_reader.cpp

```diff
@@ -395,6 +403,9 @@ CNNNetwork convert_to_cnnnetwork(std::shared_ptr<ngraph::Function>& function,
                 rtInfo.erase(it);
             }
 
+            if (need_validate_nodes_and_infer_types)
+                function->validate_nodes_and_infer_types();
```

Comment:
is it good that we re-infer the whole graph instead of just front and end parts?

Reply:
There should be an optimization ongoing, that allows not re-infer the whole graph in this method, so validate_nodes_and_infer_types() should be ok here. But I would like @GlebKazantaev to check this.

Reply:
Yes, this validation part is necessary, otherwise you can get a Function in invalid state. About optimization, we have reduced the number of Function validations which gave us significant speedup and other validation don't really take a lot of time so the generic optimization now is in on hold state.